pkg/report/output/security/security.go

```diff
@@ -534,7 +542,92 @@ func iterativeDigitsCount(number int) int {
 	return count
 }
 
-func SortResult(data []Result) {
+// removeDuplicates removes detections for same detector with same line number by keeping only a single highest severity detection
+func removeDuplicates(data map[string][]Result) map[string][]Result {
```

Comment:
I wonder if this could be simplified a lot by just using a map that stores whether a given rule id and line number have been seen before. Then if you process things in descending severity then "seen before" must be at least the same severity. WDYT?

Reply:
You can use a struct as the key: https://go.dev/play/p/LudMcVBoQ3R

Reply:
nice, will make the change